### PR TITLE
fix(ci): Make "test all" log output shorter

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

Sometimes GitHub truncates the log output of "test all", which makes it hard to find the actual failure.

This impacts some release candidate PRs, and it happens pretty often, so I just went ahead and fixed it.

Part of #5518, but we still need to fix the full sync logs.

## Solution

- Turn off `--nocapture` for "test all" in OS and Docker tests
- Split Docker "test all" job into a basic and ignored-by-default step

## Review

This is something @gustavovalverde and I discussed today.

### Reviewer Checklist

  - [x] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [x] Are the PR labels correct?
  - [x] Does the code do what the ticket and PR says?
  - [x] How do you know it works? Does it have tests?

## Follow Up Work

- Fix the full sync logs

### CODE REVIEW

**Comment by teor2345** [comment]
@mergifyio update

File: .github/workflows/continous-integration-docker.yml [hunk 1]
```diff
@@ -198,10 +198,20 @@ jobs:
...
           docker pull ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }}
-          docker run --name zebrad-tests -t ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests" --workspace -- --nocapture --include-ignored
+          docker run --name zebrad-tests -t ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests" --workspace
+
+      # Run ignored by default acceptance tests, only showing command output if the test fails.
+      #
+      # If some tests hang, add "-- --nocapture" for just that test, or for all the tests.
+      - name: Run ignored zebrad tests
+        run: |
+          docker run --name zebrad-tests -t ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests" --workspace -- --ignored
```
Comment (teor2345) [inline .github/workflows/continous-integration-docker.yml]:
```suggestion
          docker run --name zebrad-tests-ignored --tag ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests" --workspace -- --ignored
```

File: .github/workflows/continous-integration-docker.yml [hunk 2]
```diff
@@ -217,10 +227,15 @@ jobs:
...
+      - name: Run basic zebrad tests
+        run: |
+          docker pull ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }}
+          docker run --name zebrad-tests -t ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests getblocktemplate-rpcs" --workspace
+
+      - name: Run ignored zebrad tests
         run: |
           docker pull ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }}
-          docker run --name zebrad-tests -t ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests getblocktemplate-rpcs" --workspace -- --nocapture --include-ignored
+          docker run --name zebrad-tests -t ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests getblocktemplate-rpcs" --workspace -- --ignored
```
Comment (teor2345) [inline .github/workflows/continous-integration-docker.yml]:
```suggestion
          docker run --name zebrad-tests--ignored --tag ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests getblocktemplate-rpcs" --workspace -- --ignored
```

File: .github/workflows/continous-integration-docker.yml [hunk 3]
```diff
@@ -198,10 +198,20 @@ jobs:
...
 
-      - name: Run all zebrad tests
+      # Run unit and basic acceptance tests, only showing command output if the test fails.
+      #
+      # If some tests hang, add "-- --nocapture" for just that test, or for all the tests.
+      - name: Run basic zebrad tests
         run: |
           docker pull ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }}
-          docker run --name zebrad-tests -t ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests" --workspace -- --nocapture --include-ignored
+          docker run --name zebrad-tests -t ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests" --workspace
```
Comment (teor2345) [inline .github/workflows/continous-integration-docker.yml]:
```suggestion
          docker run --name zebrad-tests --tag ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests" --workspace
```

File: .github/workflows/continous-integration-docker.yml [hunk 4]
```diff
@@ -217,10 +227,15 @@ jobs:
         with:
           short-length: 7
 
-      - name: Run all zebrad tests
+      - name: Run basic zebrad tests
+        run: |
+          docker pull ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }}
+          docker run --name zebrad-tests -t ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests getblocktemplate-rpcs" --workspace
```
Comment (teor2345) [inline .github/workflows/continous-integration-docker.yml]:
```suggestion
          docker run --name zebrad-tests --tag ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} cargo test --locked --release --features "lightwalletd-grpc-tests getblocktemplate-rpcs" --workspace
```

**Comment by dconnolly** [comment]
I lost the log snapshot but I restarted the CI jobs for 'disk full' messages on the OS jobs (linux and macos)

**Review by dconnolly — APPROVED** [review]
Does not recognize `--tag`:

image

**Review by dconnolly — CHANGES_REQUESTED** [review]
Does not recognize `--tag`:

image

**Comment by teor2345** [comment]
> Does not recognize `--tag`:
> 
> image

Thanks, the options should be fixed now.

**Comment by teor2345** [comment]
Opened bug #5532 and updated bug #5494 for test failures.

**Comment by teor2345** [comment]
@mergifyio update

**Comment by teor2345** [comment]
It seems like @dconnolly is busy, @gustavovalverde would you mind reviewing this PR?

**Review by dconnolly — DISMISSED** [review]
passing, looks good

**Comment by dconnolly** [comment]
Re-running the fail: 

image

**Comment by teor2345** [comment]
@mergifyio update

**Review by dconnolly — DISMISSED** [review]

**Comment by teor2345** [comment]
I merged in the changes from PR #5549, then only applied --nocapture to the on-by-default tests.

**Review by dconnolly — APPROVED** [review]

**Comment by teor2345** [comment]
@mergifyio refresh

**Pull request merged**